### PR TITLE
Add Test Runs using Http Fault Injector

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

Adds the ability for tests to run with HTTP fault injecting functionality to simulate scenarios where the network is unstable.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by alzimmermsft** [comment]
@mikeharder, I've disabled running HTTP fault injector on macOS while a solution for it preventing requests to the HTTPS port is being found.

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/HttpFaultInjectingTests.groovy [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+import java.util.concurrent.ThreadLocalRandom
+import java.util.concurrent.atomic.AtomicInteger
+/**
+ * Set of tests that use <a href="">HTTP fault injecting</a> to simulate scenarios where the network has random errors.
+ */
+@LiveOnly
+// macOS has known issues running HTTP fault injector, change this once
+// https://github.com/Azure/azure-sdk-tools/pull/6216 is resolved
+@IgnoreIf({ OperatingSystem.current.family == OperatingSystem.Family.MAC_OS })
+class HttpFaultInjectingTests extends APISpec {
```
[the remark this answers is not in the document]
Comment (alzimmermsft) [reply sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/HttpFaultInjectingTests.groovy:34]: Agreed, I think for now we can move forward with it being local to Storage while Spock is still being used for writing tests. Once we want fault injecting testing in other SDKs, or just generally, we can add a test class into azure-core-test to make this general.

**Review by srnagar — APPROVED** [review]

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/HttpFaultInjectingTests.groovy (hunk 1 above)
[the remark this answers is not in the document]
Comment (alzimmermsft) [reply sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/HttpFaultInjectingTests.groovy:34]: https://github.com/Azure/azure-sdk-for-java/issues/35253

**Pull request merged**